api/kroxylicious-filter-api/src/main/java/io/kroxylicious/proxy/filter/KrpcFilterContext.java

```diff
@@ -57,7 +57,12 @@ public interface KrpcFilterContext {
 
     /**
      * Send a response towards the client, invoking downstream filters.
+     * <p>If this is invoked while the message is flowing downstream towards the broker, then
+     * it will not be sent to the broker. So this method can be used to generate responses in
+     * the proxy.</p>
      * @param response The response to forward to the client.
+     * @throws AssertionError if the response is not of type *ResponseData
+     * @throws AssertionError if the response apiKey does not match this context's request's apiKey
```

Comment:
is documenting the errors here too specific for the interface? Maybe it should instead have something in the paragraph above saying "The context implementation should throw exceptions if users attempt to do logically invalid operations like respond with request data or respond with data for a different apiKey" and have the specifics in the javadoc for DefaultFilterContext